proxy/envoy/route.go

```diff
@@ -316,14 +316,16 @@ func sharedHost(parts ...[]string) []string {
 	}
 }
 
-func buildTCPRoute(cluster *Cluster, addresses []string) *TCPRoute {
+func buildTCPRoute(cluster *Cluster, addresses []string, ports string) *TCPRoute {
 	// destination port is unnecessary with use_original_dst since
 	// the listener address already contains the port
 	route := &TCPRoute{
-		Cluster:    cluster.Name,
-		clusterRef: cluster,
+		Cluster:           cluster.Name,
+		DestinationIPList: make([]string, 0, len(addresses)),
+		DestinationPorts:  ports,
```

Comment:
Does tcp route have multiple ports?

Reply:
It can. But we are only doing single port right now.